crates/bevy_reflect/src/enums/enum_trait.rs

```diff
@@ -0,0 +1,273 @@
+use crate::{DynamicEnum, Reflect, VariantInfo, VariantType};
+use bevy_utils::HashMap;
+use std::any::{Any, TypeId};
+use std::borrow::Cow;
+use std::slice::Iter;
+
+/// A trait representing a [reflected] enum.
+///
+/// This allows enums to be processed and modified dynamically at runtime without
+/// necessarily knowing the actual type. Enums, unlike their struct counterparts,
+/// are a lot more complex. Users will need to be mindful of conventions,
+/// considerations, and complications when working with this trait.
+///
+/// # Variants
+///
+/// An enum is a set of choices called _variants_. An instance of an enum can only
+/// exist as one of these choices at any given time. Consider Rust's [`Option<T>`].
+/// It's an enum with two variants: [`None`] and [`Some`]. If you're `None`, you can't
+/// be `Some` and vice versa.
+///
+/// > ⚠️ __This is very important:__
+/// > The [`Enum`] trait represents an enum _as one of its variants_.
+/// > It does not represent the entire enum since that's not true to how enums work.
+///
+/// Variants come in a few [flavors](VariantType):
+///
+/// | Variant Type | Syntax                         |
+/// | ------------ | ------------------------------ |
+/// | Unit         | `MyEnum::Foo`                  |
+/// | Tuple        | `MyEnum::Foo( i32, i32 )`      |
+/// | Struct       | `MyEnum::Foo{ value: String }` |
+///
+/// As you can see, a unit variant contains no fields, while tuple and struct variants
+/// can contain one or more fields. The fields in a tuple variant is defined by their
+/// _order_ within the variant. Index `0` represents the first field in the variant and
+/// so on. Fields in struct variants, on the other hand, are represented by a _name_.
```

Comment:
Is this true for tuple structs too? If so, that deserves a bit of a note.

Reply:
It's not true of `TupleStruct` as they also use the index order. I updated the comment.